### PR TITLE
create POST /community-events

### LINKED ISSUE #5: API for creating new events
# Depends on
-  #4

### PR DESCRIPTION
**Pull request by 1234tgk**
Resolves #5 

Example request:
Screenshot 2023-06-23 at 6 50 18 PM

### CODE REVIEW

**Review by 0916dhkim — COMMENTED** [review]
Looks good overall! Just a few minor fixes and I think we're ready to merge.

File: server/src/modules/community-event/router.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import controller from './controller';
+
+const router = Router();
+
+router.route('/').post(controller.create);
```
[the remark this answers is not in the document]
Comment (1234tgk) [reply server/src/modules/community-event/router.ts:6]: @0916dhkim No. Do you prefer the latter option?

[the remark this answers is not in the document]
Comment (0916dhkim) [reply server/src/modules/community-event/router.ts:6]: I didn't know about the first option til now, so no real opinion. Let's go with the second option to be consistent with the rest of the repo.

File: server/src/modules/community-event/serializer.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      'venueContactName',
+      'venueContactPhone',
+      'venueContactEmail',
+      'announcementPosted',
+      'signUpFormSent',
+      'volunteersNeeded',
+      'voluneerRequestsSent',
+    ];
+    const fieldMap: CommunityEventResponse = fields.reduce(
+      (map: { [key: string]: any }, field) => {
```
[the remark this answers is not in the document]
Comment (1234tgk) [reply server/src/modules/community-event/serializer.ts:24]: @0916dhkim Will take a look along with [this comment](https://github.com/EOT-Event-Organizer-s-Toolbox/eot-codebase/pull/102#discussion_r1240484917). I may have to pre-populate `fieldMap` before running `reduce`, and that might be a better approach.

File: server/src/modules/community-event/types.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    'typeId' | 'organizerId' | 'announcementPosted' | 'volunteersNeeded'
+  >,
+  {
+    organizer: string;
+    eventAnnounced: boolean;
+    numVolunteersNeeded: number;
+  }
+>;
+
+export type CreateParams = {
```
Comment (anjolaoluwaakindipe) [inline server/src/modules/community-event/types.ts:25]: I think Prisma generates useful crud operation types for each model so you don't technically need to create this from scratch. 

```typescript
import { Prisma} from '@prisma/client';

export type CreateParams = Prisma.CommunityEventCreateInput
```

Comment (1234tgk) [reply server/src/modules/community-event/types.ts:25]: @anjolaoluwaakindipe This is very nifty feature from Prisma that I did not know! Thank you for pointing it out! 👍

Comment (anjolaoluwaakindipe) [reply server/src/modules/community-event/types.ts]: @1234tgk Your welcome. I found that out when the backend was still a nestjs project.

**Review by 0916dhkim — APPROVED** [review]
LGTM 👍

**Pull request merged**